mat64/dense.go

```diff
@@ -379,6 +379,27 @@ func (m *Dense) TCopy(a Matrix) {
 	*m = w
 }
 
+func (m *Dense) T() {
+	// TODO(jonlawlor): use a fully in-place transpose, without
+	// creating another slice with the same size.
+	// create a new slice
+	d := make([]float64, len(m.mat.Data))
```

Comment:
This potentially allocates to much. When m is a view on another matrix that is wider there are elements that are not used in the resulting matrix data slice.